sklearn/tests/test_common.py

```diff
@@ -87,13 +88,20 @@ def _tested_estimators():
 
 
 @parametrize_with_checks(_tested_estimators())
-def test_estimators(estimator, check):
+def test_estimators(estimator, check, request):
     # Common tests for estimator instances
     with ignore_warnings(category=(FutureWarning,
                                    ConvergenceWarning,
                                    UserWarning, FutureWarning)):
         _set_checking_parameters(estimator)
-        check(estimator)
+        check_valid_args = [p.name
+                            for p in signature(check).parameters.values()]
+        args = {}
+        if "request" in check_valid_args:
+            # pass the pytest request fixture, so that we can mark the
+            # check as XFAIL when necessary.
+            args['request'] = request
+        check(estimator, **args)
```

Comment:
What about
```python
try:
    check(estimator, request=request)
except TypeError:
    check(estimator)
```
?

Reply:
Checking the signature shouldn't be that bad. the problem with the try except , is that it will mess with the error traceback when the exception happen in the exept branch (as it will tell you that 1 exception happened in another one).

Reply:
But you are right that this can be simplified, pushed another commit for it.